packages/utils/test/loki.test.js

```diff
@@ -0,0 +1,147 @@
+import test from 'ava'
+import { Request } from '@web-std/fetch'
+import { Logging } from '../src/loki.js'
+
+function logging() {
+  return new Logging(
+    new Request('http://localhost'),
+    { waitUntil: async () => {}, passThroughOnException: () => {} },
+    {
+      url: 'test',
+      worker: 'test',
+      branch: 'test',
+      commit: 'test',
+      version: 'test',
+      env: 'test',
+    }
+  )
+}
+
+function loggingWithFilteredFields() {
+  return new Logging(
+    new Request('http://localhost'),
+    { waitUntil: async () => {}, passThroughOnException: () => {} },
+    {
+      url: 'test',
+      worker: 'test',
+      branch: 'test',
+      commit: 'test',
+      version: 'test',
+      env: 'test',
+      filterFields: (log) => {
+        const { metadata } = log
+        // Customize which fields to include
+        const filteredMetadata = {
+          request: {
+            url: metadata.request.url,
+            method: metadata.request.method,
+          },
+          response: metadata.response,
+        }
+        return { ...log, metadata: filteredMetadata }
+      },
+    }
+  )
+}
+
+test('should add a log to the batch ', async (t) => {
+  const log = logging()
+
+  log.log('testing', 'info')
+
+  t.is(log.logEventsBatch[0].level, 'info')
+  t.is(log.logEventsBatch[0].message, 'testing')
+})
+
+test('should not log with time', async (t) => {
+  const log = logging()
+
+  log.time('testing', 'description')
+
+  t.is(log.logEventsBatch.length, 0)
+  t.is(log._times.get('testing').name, 'testing')
+  t.is(log._times.get('testing').description, 'description')
+  t.assert(Date.now() >= log._times.get('testing').start)
+})
+
+test('should not log with timeend', async (t) => {
+  const log = logging()
+
+  log.time('testing')
+  await sleep(2000)
```

Comment:
does this need sleep for this long? it would be helpful to have a comment here explaining why the sleep is necessary, and I'd dial it down to 100-500 ms if feasible

Reply:
Sure, I'll reduce it to the minimum possible for easier differentiation in the test.